### ISSUE TITLE
Double Double Quote ?

### ISSUE THREAD

**Issue by Thierry-L** [uniVocity/univocity-parsers#130]
Hello,
I have an excel file that is saved in CSV format.
One field contains free 'comment' text. So some user put new lines into it as well as ".

Using UNIVOCITY in Spark, the New Line are correctly parsed : Excel quoted them with "

My problem si when this field also contains a " in the text. Excel doubles it, which seems legit as far as I understand the CSV norm

So I have something like : 

field1; field2; " this is a ""comment""
 field with
new lines"

(Yes I'm french for some obscure reason excel in French uses the ';' instead of ',' for CSV... I hope that does not put further difficulties !)

I would expect to see field1 then field2 then **this is a "comment"(\n) field with (\n) new lines**
(I put (\n) here to display the new lines in the field.

But Univocity seemed fooled and gives me :
field1, field2  AND this is a "comment

and fails with the second double double quote that is followed by a NEW LINE... 

I can provide you the exact line in Private message if you want. 

		DataFrame df = sqlContext.read().format("com.databricks.spark.csv").option("delimiter", ";")
				.option("inferSchema", "true").option("header", "true").option("charset", "windows-1252")
				.option("parserLib", "UNIVOCITY").load(nomFichierMetaClasseApplicationCarat);

Is there some option somewhere for double double quoting ?

**Comment by Thierry-L** [comment]
Re-Reading Univocity documentation I have a doubt about the ""
Is it really a propre behaviour af Excel to have use them inside a field ?
Are they restricted to 'quote' a whole field ? Should excel have put \" instead if the "is around one word in the field but not around the whole field ?

**Comment by jbax** [comment]
Interesting, I assume the defaults used by spark-csv are not the same as the defaults used by univocity-parsers. For example, the following code:

```
public static void main(String ... args){
		String in = "field1; field2; \" this is a \"\"comment\"\"\n" +
				"field with\n" +
				"new lines\"";

		CsvParserSettings s = new CsvParserSettings();
		s.getFormat().setDelimiter(';');
		String[] row = new CsvParser(s).parseLine(in);

		for(String value : row){
			System.out.println(value);
			System.out.println("----------------");
		}
	}
```

Prints this:

```
field1
----------------
field2
----------------
 this is a "comment"
field with
new lines
----------------
```

Which is what you expect. From the output you got, it seems that by default the `ignoreLeadingWhitespace` setting is switched off. There's a space in front of the opening quote of your sample input, and if `ignoreLeadingWhitespace=false` the parser assumes the whitespace is part of the value of a unquoted field. This makes it read everything until if finds either a delimiter or end of line.

Unless you are with a very old version, I can assure you the library does its job correctly. You should probably seek help from the spark-csv guys.

Thank you for using our parser!

**Comment by Thierry-L** [comment]
Thanks for your answer.
I fear the leading space was my fault in writing the test case here.

My real file has no white space. It works fine using directly the latest univocity parser.
So maybe it's something that has been fixed between 1.5.1 and 2.3.0... 
Or different parameters un spark-csv... 

Thanks !
